partner_compassion/models/partner_segment.py

```diff
@@ -0,0 +1,53 @@
+##############################################################################
+#
+#    Copyright (C) 2021 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Jonathan Guerne <jonathan.guerne@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import fields, models
+
+
+class ResPartnerSegment(models.Model):
+    _name = "res.partner.segment"
+    _description = "Partner Segmentation"
+
+    segment_index = fields.Integer(string='Segment Index used for segment computation by segmentation engine')
+    name = fields.Char(string='Name')
+    main_driver = fields.Char(string='Main driver', help='short description of the segment')
+
+    segment_size = fields.Integer(
+        help='number of partners for whom this is the primary segment',
+        compute='_compute_segment_size'
+    )
+
+    segment_total = fields.Integer(
+        help='percentage of categorized partners for whom this is the primary segment',
+        compute='_compute_segment_total'
+    )
+
+    survey_result = fields.Html()
+    image = fields.Binary(help='segment illustration')
+
+    primary_partners = fields.One2many('res.partner', compute='_compute_primary_partners')
+    secondary_partners = fields.One2many('res.partner', compute='_compute_secondary_partners')
+
```

Comment:
We can add `sql_constraints` to add some data validation:
```suggestion

_sql_constraints = [("unique_index", "unique(segment_index)", "Segment index must be unique")]
```